### DISCUSSION TITLE
mitmproxy authenticating directly to the Target web server

### DISCUSSION THREAD

**Discussion by jurekoleksiewicz** [mitmproxy/mitmproxy#6742] [General]
Hi,
there in mitmproxy help I've found that's possible to pass
  --upstream-auth
which performs Basic authentication to the next proxy in chain.

I'm wondering if there is any possibility to make mitmproxy performing automated authentication to target WEB server. For simplicity, let's assume we have just one application "behind" mitmproxy which utilizes NTML auth ... so, I'm wondering if there is an option that everybody utilizing mitmproxy (running in transparent mode) would be automatically signed-in?

Regards, Jurek.

**Comment by mhils** [comment]
There's no builtin option for that, but I suppose you could write a custom addon that does this. :)

**Reply by jurekoleksiewicz** [reply]
Maximilian,
Is there any chance you'd shared with me any similar addon .. ?
